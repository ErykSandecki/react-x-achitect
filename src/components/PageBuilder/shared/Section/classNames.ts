export const className = 'Section';

export const classNames = {
  [className]: className,
  label: `${className}__label`,
};
