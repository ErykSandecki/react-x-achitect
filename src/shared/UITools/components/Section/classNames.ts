export const className = 'Section';

export const classNames = {
  [className]: className,
  header: `${className}__header`,
  label: `${className}__label`,
  buttons: `${className}__buttons`,
};
