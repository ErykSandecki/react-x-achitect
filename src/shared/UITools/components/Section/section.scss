.Section {
  @include theme('neutral-3', border-bottom, '1px solid');

  padding: 8px 8px 16px 16px;

  &__label {
    @include theme('neutral-1', color);

    align-items: center;
    display: flex;
    font-size: 11px;
    height: 30px;
  }
}
