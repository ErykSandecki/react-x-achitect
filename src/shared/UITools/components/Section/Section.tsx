import { FC, ReactNode } from 'react';

// components
import Box from '../../..//UI/components/Box/Box';
import { Small } from '../../../UI/components/Typography';

// hooks
import { useTheme } from 'hooks';

// others
import { className, classNames } from './classNames';

// styles
import styles from './section.scss';

// types
import { E2EAttribute } from 'types';

export type TSectionProps = {
  children: ReactNode;
  label?: string;
};

export const Section: FC<TSectionProps> = ({ children, label }) => {
  const { classNamesWithTheme, cx } = useTheme(classNames, styles);

  return (
    <Box
      classes={{ className: cx(classNamesWithTheme[className]) }}
      e2eAttribute={E2EAttribute.section}
    >
      {label && (
        <Small classes={{ className: cx(classNamesWithTheme.label) }}>
          {label}
        </Small>
      )}
      {children}
    </Box>
  );
};

export default Section;
