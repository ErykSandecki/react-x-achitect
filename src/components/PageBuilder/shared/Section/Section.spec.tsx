import { render } from '@testing-library/react';

// components
import Section from './Section';

describe('Section snapshots', () => {
  it('should render Section', () => {
    // mock

    // before
    const { asFragment } = render(<Section label="label">children</Section>);

    // result
    expect(asFragment()).toMatchSnapshot();
  });
});
