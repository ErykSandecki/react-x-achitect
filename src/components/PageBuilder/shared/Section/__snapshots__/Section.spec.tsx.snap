// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section snapshots should render Section 1`] = `
<DocumentFragment>
  <div
    class="css-q0izro Section Section--light"
    data-test-section=""
  >
    <small
      class="css-q0izro Section__label Section__label--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
      data-test-text=""
      style="text-align: inherit;"
    >
      label
    </small>
    children
  </div>
</DocumentFragment>
`;
